airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -64,7 +64,8 @@ class SimpleRetriever(Retriever):
     def __post_init__(self, parameters: Mapping[str, Any]) -> None:
         self._paginator = self.paginator or NoPagination(parameters=parameters)
         self._last_response: Optional[requests.Response] = None
-        self._records_from_last_response: List[Record] = []
+        self._last_page_size: int = 0
```

Comment:
discussed a bit w/ Alex over Slack, since we yield from the previous set of records before calculating the `next_page_token`, we should have the up to date count of records when we pass `self._last_page_size` to the paginator